api/routes/auth.py

```
@@ -147,7 +143,12 @@ async def google_authorized(request: Request) -> RedirectResponse:
     try:
         google = _get_provider_client(request, "google")
         token = await google.authorize_access_token(request)
-        user_info = token.get("userinfo")
+        resp = await google.get("userinfo", token=token)
+        if resp.status_code != 200:
+            logging.warning("Failed to retrieve user info from Google")
+            raise HTTPException(status_code=400, detail="Failed to get user info from Google")
```

Ruff (0.12.2)

Abstract `raise` to an inner function

(TRY301)